inc/performance_optimizations/namespace.php

```diff
@@ -17,7 +15,11 @@ function bootstrap() {
  *
  */
 function increase_set_time_limit_on_async_upload() {
-	if ( ini_get( 'max_execution_time' ) < 120 ) {
+	if (
+		isset( $_SERVER['REQUEST_URI'] ) &&
+		strpos( $_SERVER['REQUEST_URI'], '/wp-admin/async-upload.php' ) !== false &&
+		ini_get( 'max_execution_time' ) < 120
+	) {
```

Comment:
Can we flip this and return early to avoid the giganto-if?

Reply:
Not sure if this update is quite what you had in mind. There's a lot of stuff to check either way.